### ISSUE TITLE
ChefSpec matcher not working correctly on new custom resource

### ISSUE THREAD

**Issue by nshemonsky** [chef-cookbooks/chef_nginx#38]
### Cookbook version
4.0.0

### Chef-client version
12.15.19

### Platform Details
OS X 10.11.6

### Scenario:
The current matchers do not allow for successfully ChefSpec testing the nginx_site resource. 

### Steps to Reproduce:
Attempt to spec the resource, `expect(chef_run).to enable_nginx_site('my_site')`

### Expected Result:
Successfully test the resource.

### Actual Result:
Testing results in having an untouched resource  chef_nginx_site['my_site'].

Full output can be found at this [gist](https://gist.github.com/nshemonsky/16fa4ed89ee9039ac4fcd05d068a2c0e).  In local testing I was able to successfully re-define the matchers as follows:

```
  def enable_nginx_site(name)
    ChefSpec::Matchers::ResourceMatcher.new(:nginx_site, :enable, name)
  end

  def disable_nginx_site(name)
    ChefSpec::Matchers::ResourceMatcher.new(:nginx_site, :disable, name)
  end
```

I'm not sure if that will break things for anyone still using the enable/disable properties but I don't believe it will. Happy to submit the PR if the proposed matchers are sufficient.

**Comment by wduncanfraser** [comment]
Saw this on my end as well.

**Comment by tas50** [comment]
This should be fixed now in the 4.0.1 release. Give it a try and let me know if it's not working still.
